src/api/materials.py

```
@@ -0,0 +1,541 @@
+        return w.err("Failed to fetch materials", 500)
+
+    materials = []
+    for row in res.results or []:
+        materials.append({
+            "id":          row.id,
+            "activity_id": row.activity_id,
+            "title":       row.title,
+            "description": row.description or "",
+            "uploaded_by": row.uploaded_by,
+            "created_at":  row.created_at,
+        })
```

Ruff (0.16.1)

[warning] Use a list comprehension to create a transformed list

(PERF401)